### ISSUE TITLE
Blinds - Motor controls

### ISSUE THREAD

**Issue by sy5t3ch** [homebridge-nhc2/homebridge-nhc2#31]
Hi,

I don't seem to get my blinds (motor controls) in Home app. (Blinds are working fine through the Niko home app)
Hobby password is OK. The IP is also the correct one of the smarthub.

In the logging I don't see much:

[09/02/2021, 22:37:13] [NHC2] Initializing NHC2 platform...
[09/02/2021, 22:37:13] [NHC2] Suppressing accessories: 
[09/02/2021, 22:37:13] [NHC2] NHC2Platform finished initializing!
[09/02/2021, 22:37:13] [NHC2] NHC2Platform 'didFinishLaunching'

Is it because blinds are not supported yet in this plugin? If it should be, what can I do to check what my problem is?

Thanks in advance for any advice!

**Comment by christophevg** [comment]
I have been working on support for it: https://github.com/homebridge-nhc2/homebridge-nhc2/tree/feature/sunblinds.
I was a bit stuck: #24 
I hope to finalise _some_ support soon - it's slowly getting to the top of my TODO stack ;-)

**Comment by sy5t3ch** [comment]
Thanks for your feedback and development! Looking forward to it 👍

**Comment by rubencallewaert** [comment]
Hi @christophevg,

Just started experimenting and it's very nice to see lights and dimmers work perfectly.
I'm really looking forward to getting blinds working as well.

You said you were stuck with #24 , do you need any help? What's the status now?

**Comment by christophevg** [comment]
Hi @rubencallewaert,

as the problem is inherent to the different way blinds are addressed by HomeKit and NHC, I won't be possible to have 100% support. So I decided to move forward with reduced support, starting with "simply" opening and closing, omitting feedback and open/close notion. So I picked up development again, starting with additions to the nhc2-hobby-api. But ran in some issues there. I've pinged @wvanvlaenderen on [Gitter](https://gitter.im/homebridge-nhc2/community) (once more here now ;-)) to see how to progress with this. Once the additional properties and actions are in there, I'll add basic motor support to the plugin. Afterwards, we can see if this is enough (for most people), or if we need additional effort to get to a higher level of support.

It's nearly at the top of my TODO stack, popping up more often now, so with some luck, I hope to cook something up rather soon ;-)

**Comment by rubencallewaert** [comment]
Hi @christophevg did you have a chance to look at my 3 open PR's already 1 here and 2 in https://github.com/homebridge-nhc2/nhc2-hobby-api ?
They should wrap up blinds support as good as we can for now.

**Comment by christophevg** [comment]
No, not yet myself. The coming days are also not really open to experimenting :-( @wvanvlaenderen follows up and maintains the master branch + releases. So, with a bit of luck, we'll all enjoy blinds support pretty soon.

**Comment by wvanvlaenderen** [comment]
Sorry y'all, been quite busy lately. I'll try to wrap this up by the end of the week.

**Comment by jdecupere** [comment]
First of all: thank you all for your work and making this HB plug-in. 
I am eagerly awaiting this update as now I can only control my screens with a workaround in NHC itself (assigning a switch which is then available in HK through this HB plugin).
Anyway: What I wanted to add here is that I received an email from Niko on their latest update and it includes:
_De status van je toestellen bekijken: Is je garagepoort open, gesloten of beweegt ze nog? En je rolluiken of zonneschermen? Een extra omschrijving geeft duidelijk de status van je geconnecteerde toestellen aan._
Does this mean some of the restrictions you guys are talking about now is gone?

**Comment by christophevg** [comment]
@jdecupere I'm afraid the update isn't at NHC2 protocol level, but rather at UI level in the Niko app, which now has additionally the "extra description" below the icons:
![IMG_EE0A90AC0C14-1](https://user-images.githubusercontent.com/124216/115429472-b928ea00-a203-11eb-862c-8fe3c8ad6cc8.jpeg)
I already had the impression that was "new" 😉 And the mailing today seems to confirm that. For a second I also was hoping when I read it 🤣

**Comment by rubencallewaert** [comment]
@wvanvlaenderen gentle reminder 😉.

FYI: If needed, I'm willing to become a maintainer on the repo's to add features and help get releases published.

After we get blinds working, I'm looking forward to getting thermostats working as well.

**Comment by wvanvlaenderen** [comment]
@rubencallewaert I didn't find the time to look into this yet. I've invited you to become a maintainer, this should speed stuff up. Let's get in touch on https://gitter.im/homebridge-nhc2/community to align on the release process.

**Comment by intivalderas** [comment]
Any update on the blinds implementation?

**Comment by christophevg** [comment]
@wvanvlaenderen @rubencallewaert Can we quickly agree on what still needs to be done to get this feature into the released version? It has surfaced my personal TODO list also again, so I'm willing to put in the effort to finalise this.

**Comment by rubencallewaert** [comment]
Normally I have 3 changes that should deliver this feature in the best way possible we have right now. (Niko Hobby API limitations regarding the moving direction state).

If we merge these 3 PR's this feature should be working:

homebridge-nhc2/nhc2-hobby-api#17
homebridge-nhc2/nhc2-hobby-api#18
homebridge-nhc2/homebridge-nhc2#33

**Comment by PieterMoulaert** [comment]
That would be great, I could help test this out if that would aid you.

**Comment by nycho1986** [comment]
I was wondering if there was any update on this feature 😬.

**Comment by rubencallewaert** [comment]
@nycho1986 See the following comment (Project is moving to new org: https://github.com/openhomekit)

https://github.com/homebridge-nhc2/homebridge-nhc2/pull/33#issuecomment-1152630100
